File: frontend/src/objects/merchant/menu/MenuItemSelectionGroup.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type { DisplayText, EntityCount } from '@/objects/core/SharedObjects'
import type { MenuItemSelectionOption } from '@/objects/merchant/menu/MenuItemSelectionOption'

export type MenuItemSelectionGroup = {
  name: DisplayText
  minSelections: EntityCount
  maxSelections: EntityCount
  options: MenuItemSelectionOption[]
}

File: backend/src/main/scala/services/merchant/objects/MenuItemSelectionGroup.scala
package services.merchant.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class MenuItemSelectionGroup(
    name: DisplayText,
    minSelections: EntityCount,
    maxSelections: EntityCount,
    options: List[MenuItemSelectionOption],
)
object MenuItemSelectionGroup:
  given Encoder[MenuItemSelectionGroup] = deriveEncoder
  given Decoder[MenuItemSelectionGroup] = deriveDecoder
